docs/book/src/developer/logging.md

```diff
@@ -0,0 +1,148 @@
+# Logging
+The Cluster API project is committed to improving the SRE/developer experience when troubleshooting issues, and logging
+plays an important part in this goal.
+
+In Cluster API we strive to follow three principles while implementing logging:
+
+- **Logs are for SRE & developers, not for end users!**
+  Whenever an end user is required to read logs to understand what is happening in the system, most probably there is an
+  opportunity for improvement of other observability in our API, like e.g. conditions and events.
+- **Navigating logs should be easy**:
+  We should make sure that SREs/Developers can easily drill down logs while investigating issues, e.g. by allowing to
+  search all the log entries for a specific Machine object, eventually across different controllers/reconciler logs.
+- **Cluster API developers MUST use logs!**
+  As Cluster API contributors you are not only the ones that implement logs, but also the first users of them. Use it!
+  Provide feedback!
+
+## Upstream Alignment
+
+Kubernetes defines a set of [logging conventions](https://git.k8s.io/community/contributors/devel/sig-instrumentation/logging.md), 
```

Comment:
https://github.com/kubernetes-sigs/cluster-api/pull/6969/files/91ccef7983b99f93bdaa30f8846f0c29e46dbbb3#diff-f519e7a9f2f1e96a5cbee2b82026a0de78a1836f889153e485e6581a7ad83e92 required to remove a similar note.
Also, we have specific pointers to Kubernetes guidelines all around the doc, so I don't think that having something super generic here adds much value.

Reply:
Should we just drop the `Upstream Alignment` section? I guess it should be fine since we are already referencing the upstream guidelines at various places in the document.